cnap/core/engines/tensorflow_engine.py

```diff
@@ -0,0 +1,402 @@
+
+    Args:
+        input_size (Tuple[int, int]): Expected input size of the model.
+        dtype (str): Data type of the model.
+    """
+    def __init__(self, input_size: Tuple[int, int], dtype: str):
+        self._input_size = input_size
+        self._dtype = dtype
+
+    def preprocess(self, frame: np.ndarray) -> np.ndarray:
+        """
+        Preprocess the input frame by resizing and converting to the appropriate data type.
+
+        Args:
+            frame: An np.ndarray object representing the input frame.
+
+        Returns:
+            An np.ndarray object representing the preprocessed frame.
+        """
+        if frame.shape[2] != 3:
+            raise ValueError("Input frame should have 3 channels (BGR)")
+
+        input_height, input_width = self._input_size
+
+        if input_height is None or input_width is None:
+            resized_frame = frame
+        else:
+            resized_frame = cv2.resize(frame, (input_width, input_height))
+
+        resized_frame = cv2.cvtColor(resized_frame, cv2.COLOR_BGR2RGB)
+        batch_frame = np.expand_dims(resized_frame, axis=0)
+
+        input_tensor = tf.convert_to_tensor(batch_frame, dtype=tf.as_dtype(self._dtype))
+
+        return input_tensor.numpy()
+
+class TensorFlowPostprocessor(Postprocessor):
+    """
+    A class for postprocessing output data for TensorFlow models.
+
+    Args:
+        target (str): Target task of the model.
+        drawing (dict): Settings for drawing the results on the output frame.
+    """
+    def __init__(self, target: str, drawing: dict):
+        self._postprocessor = self._select_postprocessor(target, drawing)
+
+    def _select_postprocessor(self, target: str, drawing: dict) -> Postprocessor:
+        """
+        Select the appropriate postprocessor for the inference task.
+
+        Args:
+            target: A str object representing the target of the inference task.
+            drawing: A dict object representing the drawing configuration of the inference task.
+
+        Returns:
+            A Postprocessor object representing the appropriate postprocessor for the inference task.
+
+        Raises:
+            ValueError: If the target is not supported.
+        """
+        if target == 'object-detection':
+            return ObjectDetectionPostprocessor(drawing)
+        if target == 'face-recognition':
+            return FaceRecognitionPostprocessor(drawing)
+
+        raise ValueError(f"Unsupported inference target: {target}")
+
+    def postprocess(self, frame: np.ndarray, outputs: dict) -> np.ndarray:
+        """
+        Postprocess the output data by drawing the results on the frame.
+
+        Args:
+            frame: An np.ndarray object representing the input frame.
+            outputs: A dict object representing the output data.
+
+        Returns:
+            An np.ndarray object representing the postprocessed frame.
+        """
+        return self._postprocessor.postprocess(frame, outputs)
```

Comment:
Do we need to specify the `KeyError` and `ValueError` raised by https://github.com/intel/cloud-native-ai-pipeline/blob/main/cnap/core/processors/postprocessor.py#L72-L75 in docstring?